Connected System Plug-in (CSP) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/googledrive/templates/GoogleDriveUploadFileIntegrationTemplate.java

```diff
@@ -0,0 +1,123 @@
+package com.mycorp.googledrive.templates;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+
+import com.appian.connectedsystems.simplified.sdk.SimpleIntegrationTemplate;
+import com.appian.connectedsystems.simplified.sdk.configuration.SimpleConfiguration;
+import com.appian.connectedsystems.templateframework.sdk.ExecutionContext;
+import com.appian.connectedsystems.templateframework.sdk.IntegrationResponse;
+import com.appian.connectedsystems.templateframework.sdk.TemplateId;
+import com.appian.connectedsystems.templateframework.sdk.configuration.Document;
+import com.appian.connectedsystems.templateframework.sdk.configuration.FolderPropertyDescriptor;
+import com.appian.connectedsystems.templateframework.sdk.configuration.PropertyPath;
+import com.appian.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.common.base.Stopwatch;
+
+@TemplateId(name = "GoogleDriveUploadFileIntegrationTemplate")
+public class GoogleDriveUploadFileIntegrationTemplate extends SimpleIntegrationTemplate {
+
+  public static final String FILE_ID_KEY = "fileId";
+  public static final String FOLDER_ID_KEY = "folderID";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath updatedProperty,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        textProperty(FILE_ID_KEY)
+            .label("File ID")
+            .instructionText("This can be found in a shareable link for a file as part of the URL, e.g. " +
+                "https://drive.google.com/open?id=<FileID>")
+            .build(),
+        FolderPropertyDescriptor.builder()
+            .key(FOLDER_ID_KEY)
+            .label("Save to Folder")
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String fileId = integrationConfiguration.getValue(FILE_ID_KEY);
+    Long folderId = integrationConfiguration.getValue(FOLDER_ID_KEY);
+
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder = IntegrationDesignerDiagnostic
```

Comment:
Not sure of the extent of the purposes for this repo, but several things seem to be repeated throughout all of these templates -- it may be good idea to create an abstract template. It may encourage good coding patterns to devs?

Reply:
we tried to keep all templates in this repo as if they were separate projects, so that they were not tied to the same build. This to avoid adding more complexity to them